cognite/neat/rules/_analysis/_base.py

```diff
@@ -0,0 +1,10 @@
+from abc import ABC, abstractmethod
+
+from cognite.neat.rules._shared import Rules
+from cognite.neat.rules.models._rules._types import ClassEntity
+
+
+class BaseAnalysis(ABC):
+    @abstractmethod
+    def subset_rules(self, desired_classes: set[ClassEntity]) -> Rules:
```

Comment:
A good use case for generics. 
```suggestion
from typing import TypeVar

T_Rules = TypeVar("T_Rules", bound=Rules)

class BaseAnalysis(ABC, Generic[T_Rules]):
    @abstractmethod
    def subset_rules(self, desired_classes: set[ClassEntity]) -> T_Rules:
```